ls1intum/Artemis
# [Exam] Client: Fingerprint & Instance ID

### Checklist

### Motivation and Context
We want to make sure that only one student is connected to Artemis during an Exam.

### Description
We use [FingerprintJS](https://github.com/Valve/fingerprintjs2) to compute a fingerprint, based on the Browser & Operating System.
Further, we store a UUID in the local storage and include it in every request.

### Steps for Testing

1. Log in to Artemis
2. Inspect HTTP Calls and find `X-Artemis-...` Headers
3. Try with a different browser and see different values.
4. Restart browser, logout/in etc. and find the same values in each browser.
